src/tables/UserTable.java

```diff
@@ -80,6 +82,49 @@ public static void updateUser(Connection conn, User user) {
         SQLHelper.execute(conn, query);
     }
 
+    /**
+     * Delete a user from the database
+     * @param conn - the connection object to the db
+     * @param username - the username to delete
+     * @param activeAdminUsername - if an admin is deleting accounts, pass his username, otherwise pass ""
+     * @return A message indicating what happened
+     */
+    public static String deleteUser(Connection conn, String username, String activeAdminUsername) {
+
+        if (username.equals(activeAdminUsername)){
+            return "You cannot delete your own account here, please go to preferences";
+        }
+
+        String countQuery = "SELECT count(*) FROM user WHERE username='" + username + "\';";
+        ResultSet resultSet = SQLHelper.executeQuery(conn, countQuery);
+
+        try {
+            if (resultSet.next()){
+                if (resultSet.getInt("Count(*)") == 1){
+                    String query = "DELETE FROM user_interests WHERE username=\'" + username + "\';";
```

Comment:
I originally thought about doing a trigger but seemed unnecessary, however cascade deletes look like they would work, I'll try and implement them.